### ISSUE TITLE
Editing image builder .preset file does not trigger asset rebuild 

### ISSUE THREAD

**Issue by michabr** [o3de/o3de#2115]
**Describe the bug**
After editing a .preset file, images using that preset are not automatically rebuilt by the Asset Processor.

**To Reproduce**
1. Open a .preset file such as UserInterface_Lossless.preset
2. Update a property such as "DestColor": "Linear" to "DestColor": "sRGB"
3. Save the .preset file
4. Open Asset Processor if not already open
5. Asset Processor does not reprocess any assets that use the modified preset, such as Button_Sliced_Normal.tif

**Expected behavior**
When modifying a .preset file, Asset Processor will rebuild any assets that use the associated preset.

**Desktop/Device (please complete the following information):**
 - Device: PC

**Comment by lawsonamzn** [comment]
In case someone wants a hint on how to fix this ,likely you'd add code to the CreateJobs function in Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp

... to declare the preset file and any other metadata files (such as imagesettings file) as a Source File Dependency by adding it to the array of them in "result" struct.

Source file dependencies do not need to exist at the time of emission but will cause rebuild when they pop into existence or change.

**Comment by santorac** [comment]
I believe it's a little more complicated than that. IIRC the image processing system loads the .preset files when the AP is started, not when running CreateJobs. We would need to do what @lawsonamzn suggests and _also_ change the way that image processing tracks the presets so they can be reloaded as needed. Maybe every time some client code tries to fetch a preset, the image system should check whether it is stale.

**Comment by lawsonamzn** [comment]
There's a number of ways to do that.  One option would be to, in create jobs, figure out which preset to use, and use a recorded timestamp of it in the builder to retrigger reload (?).  Createjobs should be really, really fast, since it runs up front and blocks startup of the editor...  its processjobs that can be slow since its done in paralell and can happen lazily.   Maybe there's a way to only check for reload of the preset in processjobs, or to just load the presets or the bulk data of them (as opposed to eg, just the name) as part of processjob instead of caching?  How long does it take to load preset data?

**Comment by VickyAtAZ** [comment]
Load a preset data is very fast. 
The use cases we have problem is some preset change (specially the "fileMask" change) will lead to change that some images which were not using this preset may start to use the preset. This kind of change can't be resolved by setting source dependency. 
But we make some change to resolve most common cases such as the images which uses the same preset before or after the preset change.

**Comment by lawsonamzn** [comment]
It is possible to establish a dependency on "*.preset" in createjobs, but I'm not sure if this causes it to recompile everything if any preset changes.  If not, it means that you could trigger createjobs for any change in any preset, but only change the fingerprint of the job itself if the actual preset changed.

**Comment by VickyAtAZ** [comment]
The main issue is we don't know which preset would be used for a specific job (texture file) and the current code of decide which preset to use can be expensive for certain cases and we won't want to move it to CreateJobs. 
Maybe we need to consider to optimize the select preset function and move it to CreateJobs.

**Comment by lawsonamzn** [comment]
Yes, its a hidden dependency.  You need the texture crunching job to rerun if the preset file changes, but it doesn't currently express a dependency between it and the preset file (so won't rerun if the preset changes). 

So without having some way to know which preset to use to compress the image at create jobs time its going to be exceedingly difficult to express that dependency properly and cause it to reprocess when the preset changes.  Even more difficult is forcing it to check when any preset changes since it could have changed what it processes by.

If it can be made quick to figure out which preset file will be used to process, it would be fairly straightforward to establish a dependency on that preset file.  However, if preset files themselves alter the images that they apply to, it means having to retrigger it on preset file change too - worse, if any preset file can cause any other texture to change what it needs to do.  You could probably do this by hashing the full set of all preset files loaded on startup and setting that as part of the analysis fingerprint for the builder, but we don't currently have an API that would mark files dirty again or recompute the analysis fingerprint for the builder.  This would require an AP restart.

**Comment by VickyAtAZ** [comment]
PR: https://github.com/o3de/o3de/pull/5639

**Comment by VickyAtAZ** [comment]
Follow up PR: https://github.com/o3de/o3de/pull/5689

ASV PR to update some screenshots: https://github.com/o3de/o3de-atom-sampleviewer/pull/283

**Comment by VickyAtAZ** [comment]
@michabr The issue is resolved in Stabilization/2110 branch. Can you verify it?

**Comment by derikson-dev** [comment]
@nick-l-o3de, I looked for UserInterface_Lossless.preset and didn't find it. @michabr states that the issue has been resolved, but needs to be verified. I Itried to test it, but I can't find the file UserInterface_Lossless.preset. Do you know something about it?
